lib/pure/strformat.nim

```diff
@@ -573,19 +587,24 @@ proc strformatImpl(pattern: NimNode; openChar, closeChar: char): NimNode =
           strlit = ""
 
         var subexpr = ""
-        while i < f.len and f[i] != closeChar and f[i] != ':':
-          if f[i] == '=':
+        var inParens = 0
+        while i < f.len and f[i] != closeChar and (f[i] != ':' or inParens!=0):
+          case f[i]
+          of '(': inc inParens
+          of ')': dec inParens
```

Comment:
```suggestion
          of ')': if inParents > 0: dec inParens
```

Reply:
i thought i'd leave (foo)) and ((foo) to be equally caught by parseExpr

Reply:
of course, i dont test for parens inside quotes either...

Reply:
```&"""({if x: ')': else: '(')}"""``` will fail

Reply:
Ok. At some point the programmer has to realize that full programs in a string format environment is bad code anyway.

Reply:
to be fair, ```&"""{'{' & "hi" & '}'}"""``` doesn't work either, even escaped as }}

options:  
  change (break) the api to use backslash escapes like a normal person, 
     at the same time implement them for \n etc doing away with the raw string gotcha?

  escape "(){}" as "{(}){{}}"? it's horrible.  "((" "))" is too confusing

  document the limitation